examples/repl.rs

```diff
@@ -70,14 +70,23 @@ mod example {
         Naive,
     }
 
+    #[derive(Debug)]
+    struct BacktraceTypeError(String);
+
+    impl std::fmt::Display for BacktraceTypeError {
+        fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+            write!(f, "Unrecognized backtrace type {}. Supported ones are 'fp' and 'naive'. Please consider using one of them.", self.0.trim())
+        }
+    }
+    impl std::error::Error for BacktraceTypeError {}
+
     impl BacktraceType {
-        fn from_str(value: &str) -> CrabResult<Self> {
+        #[inline]
+        fn from_str(value: &str) -> Result<Self, BacktraceTypeError> {
```

Comment:
I actually meant putting the body of this function in the `from_str` of the `HighlightAndComplete` impl and removing this function. Not marking this as `#[inline]`.

Reply:
But `from_str` goes with the BacktraceType implementation and not ` impl HighlightAndComplete for BacktraceType`. So, this might be the way to go

Reply:
As discussed on PM this can be resolved later.